src/main/java/com/radixdlt/crypto/HashUtils.java

```diff
@@ -0,0 +1,123 @@
+import com.radixdlt.SecurityCritical;
+import com.radixdlt.SecurityCritical.SecurityKind;
+
+import java.security.SecureRandom;
+import java.util.Comparator;
+
+@SecurityCritical(SecurityKind.HASHING)
+public final class HashUtils {
+
+	private static final Comparator<HashCode> hashComparator = new Comparator<HashCode>() {
+
+		private final Comparator<byte[]> bytesComparator = UnsignedBytes.lexicographicalComparator();
+
+		@Override
+		public int compare(HashCode o1, HashCode o2) {
+			return bytesComparator.compare(o1.asBytes(), o2.asBytes());
+		}
+	};
+
+	private static final SecureRandom secureRandom = new SecureRandom();
+
+	private static final HashHandler shaHashHandler = new SHAHashHandler();
+
+	/**
+	 * Returns a hash consisting of 32 zero bytes.
+	 */
+	public static HashCode zero256() {
+		return zero(32);
+	}
+
+	/**
+	 * Returns a hash consisting of {@code length} zero bytes.
+	 */
+	public static HashCode zero(int length) {
+		return HashCode.fromBytes(new byte[length]);
+	}
+
+	/**
+	 * Returns a random hash of length 32 bytes.
+	 */
+	public static HashCode random256() {
+		return random(32);
+	}
+
+	/**
+	 * Returns a random hash of specified length.
+	 */
+	public static HashCode random(int length) {
+		byte[] randomBytes = new byte[length];
+		secureRandom.nextBytes(randomBytes);
+		return HashCode.fromBytes(shaHashHandler.hash256(randomBytes));
+	}
+
+	/**
+	 * Hashes the supplied array, returning a cryptographically secure 256-bit hash.
+	 *
+	 * @param dataToBeHashed The data to hash
+	 * @return The digest by applying the 256-bit/32-byte hash function
+	 */
+	public static HashCode sha256(byte[] dataToBeHashed)	{
+		return sha256(dataToBeHashed, 0, dataToBeHashed.length);
+	}
+
+	/**
+	 * Hashes the specified portion of the array, returning a cryptographically secure 256-bit hash.
+	 *
+	 * @param dataToBeHashed The data to hash
+	 * @param offset The offset within the array to start hashing data
+	 * @param length The number of bytes in the array to hash
+	 * @return The digest by applying the 256-bit/32-byte hash function.
+	 */
+	public static HashCode sha256(byte[] dataToBeHashed, int offset, int length) {
+		return HashCode.fromBytes(shaHashHandler.hash256(dataToBeHashed, offset, length));
+	}
+
+	/**
+	 * Hashes the specified portion of the array, returning a cryptographically secure 512-bit hash.
+	 *
+	 * @param dataToBeHashed The data to hash
+	 * @return The 512-bit/64-byte hash
+	 */
+	public static HashCode sha512(byte[] dataToBeHashed) {
+		return HashCode.fromBytes(shaHashHandler.hash512(dataToBeHashed));
+	}
+
+	/**
+	 * Compares two HashCode instances using the underlying byte array.
+
+	 * @param fst The first object to be compared
+	 * @param snd The second object to be compared
+	 * @return A negative integer, zero, or a positive integer as the
+	 *         first argument is less than, equal to, or greater than the
+	 *         second.
+	 */
+	public static int compare(HashCode fst, HashCode snd) {

```

Comment:
What's the use case of this?

Reply:
it's used to compare pow fee against target here: `PowFeeLedgerAtomChecker` (it used `Hash.compareTo` before)

Reply:
ah, thats right. cool.